### PR TITLE
Hook up properties data source to service

### PR DESCRIPTION
**Pull request by bidetofevil**
## Goal

Hook up data source to EmbraceSessionPropertiesService so any changes to that will propagate to the session span

### CODE REVIEW

**Comment by bidetofevil** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/694?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#694** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/694?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#693** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/693?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#692** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/692?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#691** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/691?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#690** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/690?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bidetofevil and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/properties/EmbraceSessionPropertiesService.kt
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/properties/EmbraceSessionPropertiesService.kt:15]: There's a bug here where by any truncated key or value is not reflected in the session span because the validation is done at the EmbraceSessionProperties level. I will fix this tomorrow....

**Review by fractalwrench — APPROVED** [review]
LGTM if the bug around truncation is fixed (in this changeset or another) before this gets merged down to master

**Comment by bidetofevil** [comment]
I'll fix this on top of everything because it'll require moving some code around in terms of where validation is done. I'll make sure that's ready before merging this.

**Comment by bidetofevil** [comment]
Noted bug fixed in PR #700 - will merge now

**Pull request merged**
